### DISCUSSION TITLE
[Enhancement] Take some enhancements from flutter

### DISCUSSION THREAD

**Discussion by ahmednfwela** [dotnet/maui#94] [General]
## Summary
Since flutter is still proving successful to this day, i think we should take some enhancements from the flutter project.
1. Since the MVU pattern is getting adopted (which is the better strongly-typed declarative version of MVVM), I suggest having a clear architecture for developers to use in state management (like [ReactiveUI](https://reactiveui.net/) or [Provider](https://pub.dev/packages/provider) but for MVU).
2. [pub.dev](https://pub.dev/) is probably the best thing to ever happen in human history, so i would love to see nuget get some love too, and the fact that you can [depend on a package from a git repo directly](https://dart.dev/tools/pub/dependencies#git-packages) is just beautiful.
3. I realize that flutter and MAUI are inherently different, (flutter is just a plugin that only depends on native canvas controls and uses Skia SDK to draw the UI, but MAUI is a transpiler to native UI controls)
But I would love to see the same UI standards that exist in flutter
e.g. 
in flutter [everything is a widget](https://flutter.dev/docs/development/ui/widgets-intro), which greatly simplifies building custom UI controls with their own rendering mechanics.
as a result of this: animations, routing, layout design is way easier in flutter than any other UI framework

these are just some ideas on the top of my head, i might add somethings later.
please feel free to tell me what you think about this

**Comment by tomasfabian** [comment]
MVU sounds good, it reminds me Redux. 
Just please don't take this syntax:
https://flutter.dev/docs/development/ui/widgets-intro#bringing-it-all-together

It looks like an intertwined mixture of UI, presentation, business, ALL-IN-ONE logic. It's terrifying at first look...

**Comment by ahmednfwela** [comment]
@tomasfabian I think this isn't a production-grade example, it's more like how to get your hands dirty with some code

**Comment by charlesroddie** [comment]
> having a clear architecture for developers to use in state management

This should not be in the core. By all means Microsoft could recommend a library for this, if it makes users feel happier. But since .Net has generics, a good state management system does not need to know about MAUI-related types specifically, and MAUI does not need to provide a system for state management.

> i would love to see nuget get some love too... depend on a package from a git repo directly

Nuget works fine. The direct git repo suggestion is interesting. Paket and FSharp's interactive mode have that.

> in flutter everything is a widget

In Xamarin.Forms everything is a `VisualElement`.

> please feel free to tell me what you think about this

Needs more specifics about the precise differences bet. Flutter and Xamarin and how they can be addressed.

Key reasons why people might prefer Flutter over Xamarin IMO:

1. ease to get identical UI
2. the way they do things minimizes platform-dependent bugs, while the Xamarin's philosophies (everything is platform-dependent renderers, can't make breaking changes even to fix bad architecture, keep adding new features) means they haven't got on top of their bugs. I hope MAUI means some of this can change.
